packages/ipfs-http-server/src/api/resources/dag.js

```diff
@@ -198,13 +155,15 @@ exports.put = {
             throw Boom.badRequest('Failed to parse the JSON: ' + err)
           }
         } else {
-          const codec = multicodec[format.toUpperCase().replace(/-/g, '_')]
+          // the node is an uncommon format which the client should have
+          // serialized so deserialize it before continuing
+          const ipldFormat = await request.server.app.ipfs.ipld._getFormat(format)
```

Comment:
I've opened https://github.com/ipld/js-ipld/pull/292 to make this `getFormat(..)` instead of `_getFormat(..)`